datalad/distribution/siblings.py

```diff
@@ -301,6 +301,10 @@ def __call__(
     @staticmethod
     def custom_result_renderer(res, **kwargs):
         from datalad.ui import ui
+        # should we attempt to remove an unknown sibling, complain like Git does
+        if res['status'] == 'notneeded' and res['action'] == 'remove-sibling':
+            ui.message('fatal: No such remote: {name}'.format(**dict(res)))
```

Comment:
Thanks, this looks like an improvement to me.  I have the following suggestions, some nit-picks:

  * You could just use `**res` instead of passing `res` to `dict`.

  * While I think we should complain in the spirit of `git remote rm`, I think it'd be better to not complain _exactly_ like it.  In particular, I'd say we should drop "fatal: " and replace "remote" with "sibling".

  * The message should include some indication of the path so that recursive operation gives a clearer message:

    ```
    % datalad subdatasets
    subdataset(ok): subds (dataset)
    % datalad --on-failure=continue siblings -r remove -s nonexistent
    fatal: No such remote: nonexistent
    fatal: No such remote: nonexistent
    ```

Reply:
Thanks for these comments, I think I have addressed them now.